File: quicksort.c
#include<stdio.h>

int partition(int *a, int p, int r) {
  int pivot = a[r]; // pivot
  int i = p-1, j, temp;
  for(j = p; j <= r-1; j++) {
    // If current element is smaller than or equal to pivot
    if(a[j] <= pivot) {
      i++;
      temp = a[i];
      a[i] = a[j];
      a[j] = temp;
    }
  }
  temp = a[i+1];
  a[i+1] = a[r];
  a[r] = temp;
  return i+1;
}

void quick_sort(int *a, int p, int r) {
  if(p < r) {
    int q = partition(a,p,r);
    quick_sort(a, p, q-1);
    quick_sort(a, q+1, r);
  }
}

int main() {
  int n, i;
  printf("introduce el tamaño de tu arreglo \n");
  scanf("%d", &n);
  int a[n];
  printf("introduce los elementos del arreglo \n");
  for(i=0; i<n; i++) {
    scanf("%d", &a[i]);
  }
  quick_sort(a, 0, n-1);
  printf("el arreglo ordenado es: \n");
  for(i=0; i<n; i++) {
    printf("%d \n", a[i]);
  }
  return 0;
}

File: quicksort.py
import random
R = random.Random(50)

def qsort(L):
	quicksort(L,0,len(L))

def quicksort(L,start,stop):
	if stop - start < 2: return
	key = L[R.randrange(start,stop)]

	e = u = start
	g = stop
	while u < g:
		if L[u] < key:
			swap(L,u,e)
			e = e + 1
			u = u + 1
		elif L[u] == key:
			u = u + 1
		else:
			g = g - 1
			swap(L,u,g)
	quicksort(L,start,e)
	quicksort(L,g,stop)

def swap(A,i,j):
	temp = A[i]
	A[i] = A[j]
	A[j] = temp
	
print("quicksort:\t\n")
L = [3,1,4,1,5,9,2,6,5,3,5,8,9,7,9]
print("tu arreglo desordenado  era:\t\n" ,L)
qsort(L)
print ('tu arreglo ordenado es:\n' ,L)